web/pages/notifications.tsx

```diff
@@ -362,27 +365,83 @@ function IncomeNotificationItem(props: {
   justSummary?: boolean
 }) {
   const { notification, justSummary } = props
-  const { sourceType, sourceUserName, sourceUserUsername } = notification
+  const { sourceType, sourceUserName, sourceUserUsername, sourceText } =
+    notification
   const [highlighted] = useState(!notification.isSeen)
   const { width } = useWindowSize()
   const isMobile = (width && width < 768) || false
   useEffect(() => {
     setNotificationsAsSeen([notification])
   }, [notification])
 
-  function getReasonForShowingIncomeNotification(simple: boolean) {
+  function reasonAndLink(simple: boolean) {
     const { sourceText } = notification
     let reasonText = ''
     if (sourceType === 'bonus' && sourceText) {
       reasonText = !simple
         ? `Bonus for ${
             parseInt(sourceText) / UNIQUE_BETTOR_BONUS_AMOUNT
-          } unique traders`
+          } unique traders on`
         : 'bonus on'
     } else if (sourceType === 'tip') {
-      reasonText = !simple ? `tipped you` : `in tips on`
+      reasonText = !simple ? `tipped you on` : `in tips on`
+    } else if (sourceType === 'betting_streak_bonus' && sourceText) {
+      reasonText = `for your ${
+        parseInt(sourceText) / BETTING_STREAK_BONUS_AMOUNT
+      } day`
     }
-    return reasonText
+    return (
+      <>
+        {reasonText}
+        {sourceType === 'betting_streak_bonus' ? (
+          simple ? (
+            <span className={'ml-1 font-bold'}>Betting Streak</span>
+          ) : (
+            <SiteLink
+              className={'ml-1 font-bold'}
+              href={'/betting-streak-bonus'}
+            >
+              Betting Streak
```

Comment:
Ideally this would be "10-day betting streak" but this is fine for now.

Reply:
Sure, changed it